ui/app/components/license-banners.js

```diff
@@ -24,4 +44,20 @@ export default class LicenseBanners extends Component {
     if (!this.args.expiry) return 99;
     return differenceInDays(new Date(this.args.expiry), new Date());
   }
+
+  @action
+  dismissBanner(dismissAction) {
+    // updates localStorage and then updates the template by calling updateDismissType
+    localStorage.setItem(`dismiss-license-banner-${this.currentVersion}`, dismissAction);
```

Comment:
If I'm understanding this correctly, if I've previously dismissed a warning and I come back and dismiss an expired, we will replace the localStorage value so that the warning will be able to show up again?

Reply:
Correct. The key here is that both will never show up at the same time. So let's say you dismissed a warning, but then you let your license expire, the expired license will show. If we dismiss that, then next time a warning should display it will.

Reply:
Great context, thanks!